internal/infrastructure/mongo/dataset.go

```diff
@@ -69,6 +69,16 @@ func (r *datasetRepo) FindBySchema(ctx context.Context, schemaID id.DatasetSchem
 	}, pagination)
 }
 
+func (r *datasetRepo) CountBySchema(ctx context.Context, id id.DatasetSchemaID) (int, error) {
+	res, err := r.client.Count(ctx, bson.M{
+		"schema": id.String(),
+	})
```

Comment:
```suggestion
	res, err := r.client.Count(ctx, r.readFilter(bson.M{
		"schema": id.String(),
	}))
```

Filter queries to prevent access to documents that should not be visible to the current user.